tests/examples_style.conf

```diff
@@ -0,0 +1,49 @@
+# Code formatting rules for Arduino examples, taken from:
+#
+#   https://github.com/arduino/Arduino/blob/master/build/shared/examples_formatter.conf
+#
+# This configuration file contains a selection of the available options provided by the formatting tool "Artistic Style"
+# http://astyle.sourceforge.net/astyle.html
+#
+# If you wish to change them, don't edit this file.
+# Instead, copy it in the same folder of file "preferences.txt" and modify the copy. This way, you won't lose your custom formatter settings when upgrading the IDE
+# If you don't know where file preferences.txt is stored, open the IDE, File -> Preferences and you'll find a link
+
+mode=c
+
+# 2 spaces indentation
+indent=spaces=2
+
+# also indent macros
+indent-preprocessor
+
+# indent classes, switches (and cases), comments starting at column 1
+indent-classes
+indent-switches
+indent-cases
+indent-col1-comments
+
+# put a space around operators
+pad-oper
+
+# put a space after if/for/while
+pad-header
+
+# if you like one-liners, keep them
+keep-one-line-statements
```

Comment:
There was talk of replacing "keep-one-line-statements" w/"add-one-line-braces".  It's not 100% Arduino style, but it is IMO a better style as it avoids accidentally changing code flow when adding debug statements/etc.

Reply:
I agree with adding `add-one-line-braces`, but it is not the opposite of `keep-one-line-statements`. Without `keep-one-line-statements`, this:
```c++
do_foo(); do_bar();
```
will be changed to this:
```c++
do_foo();
do_bar();
```

Whereas `add-one-line-braces` will change
```c++
if (foo)
    bar();
```
into
```c++
if (foo)
    { bar(); }
```

If we *are* adding braces, i would probably go for `add-braces` instead, which would change the above into

```c++
if (foo) {
    bar();
}
```

Reply:
Personally, I prefer this:
if(condition)
  1-liner;
else
  other-1-liner;

over this:
if(condition) {
  1-liner;
} else {
  other-1-liner;
}

the reason is quite simple: it reduces complexity at a glance. But I don't consider it critical, and my glance can adapt.

Whatever else, this should be avoided:
if(condition) 1-liner;

Reply:
Ah, i see now. Arduino config already has `add-brackets` option which was renamed to `add-braces` in latest astyle (but the old name is still supported).

So we are already converting 
```c++
if (foo)
    bar();
```
into
```c++
if (foo) {
    bar();
}
```